src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.cs

```diff
@@ -6,18 +6,32 @@ namespace System.Globalization
 {
     internal static partial class GlobalizationMode
     {
+        // Order of these properties matter because GetUseIcuMode is dependent on Invariant.
+        // So we need Invariant to be initialized first.
         internal static bool Invariant { get; } = GetGlobalizationInvariantMode();
 
-        // GetInvariantSwitchValue calls CLRConfig first to detect if the switch is defined in the config file.
+#if TARGET_WINDOWS
+        internal static bool UseIcu { get; } = GetUseIcuMode();
+#else
+        // We can't just call GetUseIcuMode and return false there if we want the linker to trim out the Windows code.
+        internal static bool UseIcu => true;
+#endif
+
+        internal static bool GetInvariantSwitchValue() =>
+            GetSwitchValue("System.Globalization.Invariant", "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT");
+
+        internal static bool GetUseNlsSwitchValue() =>
+            GetSwitchValue("System.Globalization.UseNls", "DOTNET_SYSTEM_GLOBALIZATION_USENLS");
```

Comment:
Is there a reason why the config switch is called UseNls, and then we invert it and have a property called UseIcu ? It would be nice to be consistent.

Reply:
The reason why I chose that name was to tell our customers that Icu is the default, so if you want to use nls set `UseNls` explicitly to true and also to make sure the name has NLS, so that it is clear is a Windows thing. You think it would be better to name this UseIcu and then customers would turn it off instead of on?

The reason why the property is `UseIcu` was to not invert conditions on the code and so the ICU code path was always the first code path in the condition since that is the default behavior. I think having customers turn it off might make sense.

Reply:
I think using UseNls for to config switch makes sense.

I do not think it matters which code path is the first. It is fairly common for the special less common path to be first.

Reply:
So you think it would be good to rename the property to `UseNls`? Ok, I can change that if you'd like.

Reply:
Yes, I would make the config switch and the property to have the same name.

Reply:
/cc @vitek-karas

Reply:
/cc @eerhardt - basically another feature switch